src/main/java/org/opensearch/knn/index/KnnCircuitBreakerException.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index;
+
+public class KnnCircuitBreakerException extends RuntimeException {
+    public KnnCircuitBreakerException() {
```

Comment:
Might be worth removing the default constructor so the client of this class is forced to add a message

Reply:
@shatejas What do you think about adding a default message in the constructor instead of removing it entirely?

Reply:
As long as clients have enough information to understand what went wrong the default message works. 

The intent is not to throw without a proper message considering its a custom exception and might be hard for someone to figure out what went wrong

Reply:
After trying to implement the default, it doesn't work because it's extending RuntimeException which needs the super to be called first.  I'll just remove the default constructor per your original recommendation.